src/hooks/common/useInput.ts

```diff
@@ -0,0 +1,78 @@
+import { ChangeEvent, useCallback, useState } from 'react';
+import debounce from 'lodash/debounce';
+
+export interface UseInputHookType {
+  /**
+   * input state가 initializing 될 때 넣을 값입니다.
+   *
+   * @default ''
+   */
+  initialValue?: string;
+
+  /**
+   * debounce 효과를 사용할 것인지 정의합니다.
+   *
+   * @default `false`
+   */
+  useDebounce?: boolean;
+
+  /**
+   * debounce 효과를 사용할 때의 timeout 시간입니다. (단위: ms)
+   *
+   * @default `150`
+   */
+  debounceTimeout?: number;
+}
+
+type InputAcceptType = string;
+
+export default function useInput({
+  initialValue,
+  useDebounce = false,
+  debounceTimeout = 150,
+}: UseInputHookType) {
+  const [value, setValue] = useState<InputAcceptType>(initialValue ?? '');
+  const [debouncedValue, setDebouncedValue] = useState<InputAcceptType>(initialValue ?? '');
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const handleSetDebounceValue = useCallback(debounce(setDebouncedValue, debounceTimeout), []);
```

Comment:
```suggestion
const handleSetDebounceValue = useMemo(
    () =>
      debounce((value: string) => {
        setDebouncedValue(value);
      }, debounceTimeout),
    [debounceTimeout]
  );
```

저는 exhaustive deps 룰 끄는걸 기피하는 스타일이라 이렇게 작성하곤 했는데, 생각하시기에 어떤지 궁금해요!!

> 물론 수정안하셔도 됩니다!!

Reply:
👍👍👍 

저도 린트 때문에 고민하고 있었는데... 😂 
해답을 주셨네요~~!!! 바로 반영해놨어요! 감사합니다~